### PR TITLE
Rename Notifications to Reviews

### LINKED ISSUE #1151: Change notifications to reviews
See [p99K0U-1uq-p2](https://wp.me/p99K0U-1uq) for more context

- [ ] Change the content of the notifications tab to present only reviews.
- [ ] Update the title of the tab from Notifications to Reviews

### PR DESCRIPTION
**Pull request by ctarda**
Fixes #1151 

Change the Notifications Tab to Reviews.

In action:

*Note* To keep the diff as small as possible, I have not renamed the NotificationsViewController. If / when we get this PR to a point where is mergeable, I would rename it too ReviewsViewController.

## Changes
- Renamed Notifications tab to Reviews.
- Removed dynamic filters from the view controller, including the right toolbar button

## Testing
- Check that the name of the Notifications tab item has changed to Reviews. 
- Check that the title of Reviews tab has changed to Reviews
- Leave a review in the store. Check that it is presented in the Reviews tab
- Check that the Reviews tab contains only reviews
- Try to display the empty state placeholder. Check that there are no other references to Notifications

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Notifications/NotificationsViewController.swift [hunk 1]
```diff
@@ -436,26 +355,6 @@ private extension NotificationsViewController {
     /// Refreshes the Results Controller Predicate, and ensures the UI is in Sync.
     ///
     func reloadResultsController() {
-        refreshResultsPredicate()
-        tableView.reloadData()
-    }
-
-    func refreshResultsPredicate() {
-        let notDeletedPredicate = NSPredicate(format: "deleteInProgress == NO")
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Notifications/NotificationsViewController.swift]: this `notDeletedPredicate` was used in combination with the subtype predicate, do you know if we might still want to include this filter on deleted notifications?

Comment (ctarda) [reply WooCommerce/Classes/ViewRelated/Notifications/NotificationsViewController.swift]: ah, nice catch, thanks!

**Comment by ctarda** [comment]
Thanks for the review @jaclync !

I just pushed an update that seems to fix the issue when switching stores. The solution was a combination of composing the predicate, as you mentioned, plus ensuring the predicate gets reset after the `defaultSiteWasUpdated` is called, combined with other changes in the `state` that used to happen as side effects of setting some of the properties that I removed. 

Long story short, I think it should work now 😊

**Comment by ctarda** [comment]
And one more update to change the tab bar icon!

**Comment by ctarda** [comment]
Thanks @kyleaparker 

I just pushed an update to change the colors of the dot:

**Comment by ctarda** [comment]
Sorry @jaclync , I requested another review for you by mistake 😊
